src/config.rs

```diff
@@ -419,6 +420,7 @@ fn config_from_env() -> EnvConfig {
         .map(|bucket| {
             let cred_path = env::var_os("SCCACHE_GCS_KEY_PATH")
                 .map(|p| PathBuf::from(p));
+            let url = env::var("SCCACHE_GCS_CREDENTIALS_URL").ok();
```

Comment:
This seems a little weird, because we're honoring the path ahead of the URL if both are set. Can we prevent both being set here or at least warn if they both are? The code to consume things would be a bit neater if `GCSCacheConfig` had just one of a url or path in it as well.

Reply:
iiuc, the role of config is to grab all available configs set and let the consumer handle what to do with them, that's why I didn't made `cred_path` and `url` mutually exclusive. I added the warning.